### PR TITLE
Route Filter Feature Support from Express Route

### PR DESCRIPTION
**Pull request by dihan0604**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dihan0604** [comment]
Swagger Spec: 
https://github.com/Azure/azure-rest-api-specs/pull/855

**Comment by cormacpayne** [comment]
@dihan0604 it looks like the following session record was not found: `Network.Tests.Tests.RouteFilterTests\EmptyRouteFilterTest.json`

**Comment by dihan0604** [comment]
@cormacpayne , I debugged this with Abhijeet and we realize we can not have tests in SDK. My feature is enabled in brazilUS (Stage ARM), however, for the current SDK tests, we do not provide this environment. I discussed with Abhijeet and he agreed that I can add tests in PS instead of SDK. I have removed the tests from this PR. Once brazilUS environment is available in SDK,  I will add these tests back.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs [hunk 1]
```diff
@@ -62,7 +64,7 @@ public partial class ExpressRouteCircuitPeering : SubResource
...
+        public ExpressRouteCircuitPeering(string id = default(string), string peeringType = default(string), string state = default(string), int? azureASN = default(int?), int? peerASN = default(int?), string primaryPeerAddressPrefix = default(string), string secondaryPeerAddressPrefix = default(string), string primaryAzurePort = default(string), string secondaryAzurePort = default(string), string sharedKey = default(string), int? vlanId = default(int?), ExpressRouteCircuitPeeringConfig microsoftPeeringConfig = default(ExpressRouteCircuitPeeringConfig), RouteFilter routeFilter = default(RouteFilter), ExpressRouteCircuitStats stats = default(ExpressRouteCircuitStats), string provisioningState = default(string), string gatewayManagerEtag = default(string), string lastModifiedBy = default(string), string name = default(string), string etag = default(string))
```
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs]: @dihan0604 changing the order of parameters in the constructor is a breaking change

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/Access.cs [hunk 2]
```diff
@@ -13,13 +13,12 @@ namespace Microsoft.Azure.Management.Network.Models
     using Network;
 
     /// <summary>
-    /// Defines values for NetworkOperationStatus.
+    /// Defines values for Access.
     /// </summary>
-    public static class NetworkOperationStatus
+    public static class Access
```
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/Access.cs]: @dihan0604 changing the name of this class is a breaking change

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/AzureAsyncOperationResult.cs [hunk 3]
```diff
@@ -1,61 +0,0 @@
...
-    /// The response body contains the status of the specified asynchronous
-    /// operation, indicating whether it has succeeded, is in progress, or has
-    /// failed. Note that this status is distinct from the HTTP status code
-    /// returned for the Get Operation Status operation itself. If the
-    /// asynchronous operation succeeded, the response body includes the HTTP
-    /// status code for the successful request. If the asynchronous operation
-    /// failed, the response body includes the HTTP status code for the failed
-    /// request and error information regarding the failure.
-    /// </summary>
-    public partial class AzureAsyncOperationResult
```
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/AzureAsyncOperationResult.cs]: @dihan0604 removing this class is a breaking change

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@dihan0604 a few comments about breaking changes in this pull request. Since this SDK is in preview, it is fine to introduce breaking changes, but you will need to mitigate these changes in PowerShell

Also, you have a large number of commits, so you are going to need to [rebase](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md#rebasing) to clean this up

**Comment by cormacpayne** [comment]
@dihan0604 it looks like @lamchester just opened a similar PR to this one (https://github.com/Azure/azure-sdk-for-net/pull/2798). Can we try and consolidate both PR changes into one?

**Comment by dihan0604** [comment]
@cormacpayne Comments addressed

**Comment by dihan0604** [comment]
@cormacpayne , I chatted with Chester and he agrees to wait my PR get merged and he makes his PR on top of mine. This is due to my swagger changes already merged but not the SDK.

**Pull request closed without merging**